sdk/eventhub/Microsoft.Azure.EventHubs/src/Amqp/AmqpEventHubClient.cs

```diff
@@ -34,13 +25,29 @@ public AmqpEventHubClient(EventHubsConnectionStringBuilder csb)
         {
         }
 
-        private AmqpEventHubClient(EventHubsConnectionStringBuilder csb, ITokenProvider tokenProvider)
+        public AmqpEventHubClient(EventHubsConnectionStringBuilder csb, ITokenProvider tokenProvider)
             : base(csb)
         {
             this.ContainerId = Guid.NewGuid().ToString("N");
             this.AmqpVersion = new Version(1, 0, 0, 0);
             this.MaxFrameSize = AmqpConstants.DefaultMaxFrameSize;
-            this.InternalTokenProvider = tokenProvider;
+
+            if (tokenProvider != null)
+            {
+                this.InternalTokenProvider = tokenProvider;
+            }
+            else if (string.IsNullOrWhiteSpace(csb.SharedAccessSignature))
```

Comment:
Should be !string.IsNullOrWhiteSpace
next one also

Reply:
thx